packages/squiggle-lang/src/rescript/Utility/XYShape.res

```diff
@@ -376,6 +390,131 @@ module PointwiseCombination = {
       }
     `)
 
+  /*
+    This is from an approach to kl divergence that was ultimately rejected. Leaving it in for now because it may help us factor `combine` out of raw javascript soon.
+ */
+  let combineAlongSupportOfSecondArgument0: (
+    (float, float) => result<float, Operation.Error.t>,
+    interpolator,
+    T.t,
+    T.t,
+  ) => result<T.t, Operation.Error.t> = (fn, interpolator, t1, t2) => {
+    let newYs = []
+    let newXs = []
+    let (l1, l2) = (E.A.length(t1.xs), E.A.length(t2.xs))
+    let (i, j) = (ref(0), ref(0))
+    let minX = t2.xs[0]
+    let maxX = t2.xs[l2 - 1]
+    while j.contents < l2 - 1 && i.contents < l1 - 1 {
+      let someTuple = {
+        let x1 = t1.xs[i.contents + 1]
+        let x2 = t2.xs[j.contents + 1]
+        if (
+          /* if t1 has to catch up to t2 */
+          i.contents < l1 - 1 && j.contents < l2 && x1 < x2 && minX <= x1 && x2 <= maxX
+        ) {
+          i := i.contents + 1
+          let x = x1
+          let y1 = t1.ys[i.contents]
+          let y2 = interpolator(t2, j.contents, x)
+          Some((x, y1, y2))
+        } else if (
+          /* if t2 has to catch up to t1 */
+          i.contents < l1 && j.contents < l2 - 1 && x1 > x2 && x2 >= minX && maxX >= x1
+        ) {
+          j := j.contents + 1
+          let x = x2
+          let y1 = interpolator(t1, i.contents, x)
+          let y2 = t2.ys[j.contents]
+          Some((x, y1, y2))
+        } else if (
+          /* move both ahead if they are equal */
+          i.contents < l1 - 1 && j.contents < l2 - 1 && x1 == x2 && x1 >= minX && maxX >= x2
+        ) {
+          i := i.contents + 1
+          j := j.contents + 1
+          let x = x1
+          let y1 = t1.ys[i.contents]
+          let y2 = t2.ys[j.contents]
+          Some((x, y1, y2))
+        } else {
+          i := i.contents + 1
+          None
+          // (0.0, 0.0, 0.0) // for the function I have in mind, this will error out
+          // exception PointwiseCombinationError
+          // raise(PointwiseCombinationError)
+        }
+      }
+      switch someTuple {
+      | Some((x, y1, y2)) => {
+          let _ = Js.Array.push(fn(y1, y2), newYs)
+          let _ = Js.Array.push(x, newXs)
+        }
+      | None => ()
+      }
+    }
+    T.filterOkYs(newXs, newYs)->Ok
+  }
+
+  let approximatePdfOfContinuousDistributionAtPoint: (xyShape, float) => option<float> = (
```

Comment:
Write a comment above describing what this does exactly and why it's needed

Reply:
Also, this function should probably go in Continues.res, I assume? (If it's meant for continuous distributions)

Reply:
good catch! just converted it to `XtoY.linear` and tests still pass!